### ISSUE TITLE
clair-db: CVE changes for a specific image

### ISSUE THREAD

**Issue by t00f** [arminc/clair-local-scan#4]
Hi @arminc,

I am curious to understand why CVEs are changing from 1 day to another.

For instance, I ran some tests on image `nginx:1.12` specifically, and I can see the following differences since yesterday:

From `2018-01-10` tag:
```
vulnerability:cve=CVE-2005-2541
vulnerability:cve=CVE-2007-5686
vulnerability:cve=CVE-2009-4487
vulnerability:cve=CVE-2011-3374
vulnerability:cve=CVE-2011-4116
vulnerability:cve=CVE-2013-0340
vulnerability:cve=CVE-2015-9019
vulnerability:cve=CVE-2016-2779
vulnerability:cve=CVE-2016-2781
vulnerability:cve=CVE-2016-9085
vulnerability:cve=CVE-2017-1000409
vulnerability:cve=CVE-2017-13726
vulnerability:cve=CVE-2017-15232
vulnerability:cve=CVE-2017-15422
vulnerability:cve=CVE-2017-15908
vulnerability:cve=CVE-2017-16879
vulnerability:cve=CVE-2017-17512
vulnerability:cve=CVE-2017-3738
vulnerability:cve=CVE-2017-7246
vulnerability:cve=CVE-2017-8872
vulnerability:cve=CVE-2017-9937
vulnerability:level=high
```

From `2018-01-11` (currently latest) tag:
```
vulnerability:cve=CVE-2005-2541
vulnerability:cve=CVE-2007-5686
vulnerability:cve=CVE-2010-0928
vulnerability:cve=CVE-2011-3374
vulnerability:cve=CVE-2012-3878
vulnerability:cve=CVE-2013-0337
vulnerability:cve=CVE-2013-0340
vulnerability:cve=CVE-2013-4392
vulnerability:cve=CVE-2015-9019
vulnerability:cve=CVE-2016-2779
vulnerability:cve=CVE-2016-9085
vulnerability:cve=CVE-2016-9318
vulnerability:cve=CVE-2017-15232
vulnerability:cve=CVE-2017-16879
vulnerability:cve=CVE-2017-17484
vulnerability:cve=CVE-2017-17512
vulnerability:cve=CVE-2017-17973
vulnerability:cve=CVE-2017-18018
vulnerability:cve=CVE-2017-7246
vulnerability:cve=CVE-2017-8804
vulnerability:cve=CVE-2017-9937
vulnerability:level=high
```

The following CVE disappeared: 
```
vulnerability:cve=CVE-2009-4487
vulnerability:cve=CVE-2011-4116
vulnerability:cve=CVE-2017-1000409
vulnerability:cve=CVE-2017-15908
vulnerability:cve=CVE-2017-3738
vulnerability:cve=CVE-2017-8872
```

The following CVE appeared:
```
vulnerability:cve=CVE-2010-0928
vulnerability:cve=CVE-2012-3878
vulnerability:cve=CVE-2016-9318
vulnerability:cve=CVE-2013-0337
vulnerability:cve=CVE-2017-17484
vulnerability:cve=CVE-2017-17973
vulnerability:cve=CVE-2017-18018
vulnerability:cve=CVE-2017-8804
```

Does someone else noticed these changes ? Are you running some kind of curation while creating daily `clair-db` image or Is it coming from `clair-server` directly ?

Thanks,

Chris

**Comment by t00f** [comment]
Running the same with today's image gives me some differences again:

The following CVE have been removed:
```
CVE-2007-5686
CVE-2010-0928
CVE-2013-0337
CVE-2012-3878
CVE-2013-4392
CVE-2016-9318
CVE-2017-17484
CVE-2017-17973
CVE-2017-8804
```

And the following have been added:
```
CVE-2007-6755
CVE-2009-4487
CVE-2011-4116
CVE-2013-4235
CVE-2017-15422
CVE-2017-15908
CVE-2017-5969
CVE-2017-9935
CVE-2018-1000001
```

That's pretty weird :)
Waiting for your insights to understand these changes.

Thanks,

Chris

**Comment by arminc** [comment]
I am not sure what is going on here. It would be interesting to verify if the DB has the same data every day (plus some extra). This either means it does not or clair is doing something...

**Comment by t00f** [comment]
Hi @arminc,

I understand you don't do any specific filtering before creating the new image. As I don't see any script to look at how you create the clair-db image, I first want to make sure there is nothing on your side before asking coreos guys about this behavior.

**Comment by arminc** [comment]
That is correct, I am not doing anything special with the DB as you can see here: https://github.com/arminc/clair-local-scan/blob/master/.travis.yml

The process is basically:
1) Run postgres and Clair
2) Wait and keep checking until Clair says the update process is done
3) Tag the postgres container and push it to docker hub

**Comment by t00f** [comment]
@arminc, thanks again for your message.

I have changed my setup to use a new database and vulnerabilities keeps changing when doing multiple run in a row.

It seems it is not related to clair-db. I have opened a ticket on [coreos/clair](https://github.com/coreos/clair/issues/509) to see if I can get some help.

I let you decide if you want to keep track of this conversation or close it.

Again, thanks for your help.

Chris
